### PR TITLE
[CMS-873][CMS-874] Add automation to create decoupled upstream

### PR DESCRIPTION
**Pull request by jspellman814**

### CODE REVIEW

**Comment by jspellman814** [comment]
Test of the new CircleCI workflow: https://app.circleci.com/pipelines/github/pantheon-systems/wordpress-composer-managed/90/workflows/f3ef2515-287d-4823-ba71-0041d88a6339/jobs/106

Changes in the upstream after success: https://github.com/pantheon-upstreams/decoupled-wordpress-composer-managed/blob/main-873-test/composer.json, https://github.com/pantheon-upstreams/decoupled-wordpress-composer-managed/blob/main-873-test/README.md

File: devops/files/decoupledREADME.md
Comment (jazzsequence) [inline devops/files/decoupledREADME.md:1]: Is there a standard where `decoupledREADME.md` is the established filename? I feel like this could/should be something else (e.g. `decoupled.md`, `decoupled-readme.md`, etc or even `decoupled-README.md`). The weird camelCASE is bothering me. 😅

Comment (jspellman814) [reply devops/files/decoupledREADME.md:1]: I pulled this from our implementation in [pantheon-systems/drupal-composer-managed](https://github.com/pantheon-systems/drupal-composer-managed/pull/3/commits/f66dcb9c88a93ec8139ea5e1b72d45a5141d8c59), but don't have an issue with changing the name.

Comment (jazzsequence) [reply devops/files/decoupledREADME.md:1]: I think `decoupled-README.md` makes more sense and we could update it over there as well for consistency.

**Review by jazzsequence — APPROVED** [review]
LGTM but I would defer to @backlineint

**Comment by backlineint** [comment]
> Test of the new CircleCI workflow: https://app.circleci.com/pipelines/github/pantheon-systems/wordpress-composer-managed/90/workflows/f3ef2515-287d-4823-ba71-0041d88a6339/jobs/106
> 
> Changes in the upstream after success: https://github.com/pantheon-upstreams/decoupled-wordpress-composer-managed/blob/main-873-test/composer.json, https://github.com/pantheon-upstreams/decoupled-wordpress-composer-managed/blob/main-873-test/README.md

@jspellman814  Looking at the published upstream I'm noticing that there is a composer.lock here. I was under the impression that upstreams shouldn't be published with a lock file. Could be wrong though. This is different from the Drupal upstream though, and I'd expect the approach to be consistent.

**Comment by jspellman814** [comment]
@backlineint You're correct in that the lock file shouldn't be in that repo (it's addition was a mistake on my part when I initially created the repo). It'll be removed before we make the upstream available on the platform.

**Review by backlineint — COMMENTED** [review]
I was under the impression that we'd be publishing the two plugins on packagist first before integrating this. Doesn't necessarily prevent me from approving this, just different from what I was expecting.

I did add a comment about there being a lock file in the published upstream. I think we need to resolve/clarify that one before I can approve.

**Comment by CdrMarks** [comment]
@jspellman814 Can composer.lock be added to the .gitignore file or will that cause issues for customers in the future?

**Comment by jspellman814** [comment]
> @jspellman814 Can composer.lock be added to the .gitignore file or will that cause issues for customers in the future?

Yes, this will cause issues for customers in the future. When a site is spun up using the upstream, the platform commits the composer.lock file to the repository. Additionally, [we instruct customers](https://pantheon.io/docs/guides/integrated-composer/dependencies#add-a-dependency-to-an-individual-site) to commit and push this file whenever they add dependencies.

**Comment by jspellman814** [comment]
@backlineint [pantheon-systems/pantheon-decoupled](https://packagist.org/packages/pantheon-systems/pantheon-decoupled) and [pantheon-systems/pantheon-decoupled-auth-example](https://packagist.org/packages/pantheon-systems/pantheon-decoupled-auth-example) have been released on packagist, and the composer.lock file has been removed from `pantheon-upstreams/decoupled-wordpress-composer-managed`. Additional test results here: https://app.circleci.com/pipelines/github/pantheon-systems/wordpress-composer-managed/94/workflows/d35e53cd-49be-45f3-b516-dc9180ab34e1/jobs/114 and here: https://github.com/pantheon-upstreams/decoupled-wordpress-composer-managed/tree/main-873-test-v2

**Review by backlineint — APPROVED** [review]
Resulting output here: https://github.com/pantheon-upstreams/decoupled-wordpress-composer-managed/tree/main-873-test-v2 looks right to me.

**Pull request merged**
